tests/components/history_stats/test_sensor.py

```diff
@@ -1220,7 +1220,6 @@ def _fake_states(*args, **kwargs):
             ]
         },
     )
-    await hass.async_block_till_done()
```

Comment:
Removing the block till done generally isn't great since timings tend to change over time which may make it fail again.

Maybe try moving the patch below around the setup above.

Reply:
Thanks for the suggestion, it worked well (same n=10,000, 0 failures), and definitely seems more stable than my original idea. I have updated the PR.